src/Munkres.jl

```diff
@@ -67,15 +63,39 @@ function munkres!(costMat::AbstractMatrix{T}) where T <: Real
     # 3 => PRIME => primed zero
     Zs = spzeros(Int8, rowNum, colNum)
 
+    # for tracking changes per row/col of A
+    Δrow = zeros(rowNum)
+    Δcol = zeros(colNum)
+	
     # "consider a row of the matrix A;
     #  subtract from each element in this row the smallest element of this row.
     #  do the same for each row."
-    costMat .-= minimum(costMat, dims=2)
```

Comment:
this kind of operation can be very expensive if the user wants to supply a custom matrix type with a complicated setindex! operation.
thus in my use case it is very advantageous to make the algorithm fully non mutating and non copying